(WIP) (RE-5685) Use light.exe to generate MSI

jcoconnor opened the pull request:
Initial PR to create the MSI
Needs #280 Merged first

melissa commented on lib/vanagon/platform/windows.rb:
```diff
@@ -107,7 +107,11 @@ def generate_msi_package(project)
         dir_ref = "INSTALLDIR"
-        candle_flags =  "-dPlatform=#{@architecture} -arch #{@architecture} -ext WiXUtilExtension -ext WixUIExtension"
+        wix_extensions =  "-ext WiXUtilExtension -ext WixUIExtension"
+        candle_flags =  "-dPlatform=#{@architecture} -arch #{@architecture} #{wix_extensions}"
```
Did we decide we want to keep `-dPlatform=#{@architecture}` and use it? Or would it make more sense to always take advantage of erb templates and architecture set there and remove this?

melissa replied:
I'm partial to removing this so people don't get confused when they're creating their project wix files as to where this comes from/gets set

jcoconnor replied:
Will put this up as a separate ticket [(RE-6525)](https://tickets.puppetlabs.com/browse/RE-6525) as I need to investigate if there are any other implications, but it looks like a good idea.

melissa replied:
cool

jcoconnor commented:
Updated comments for "flag maintenance" and rebased.

@melissa @McdonaldSeanp Can you review asap so we can merge this up to allow us to light up the build /cc @ScottGarman

McdonaldSeanp commented:
:+1: pending travis

melissa commented on resources/windows/wix/project.wxs.erb:
```diff
@@ -26,9 +26,6 @@
     </Feature>
 
-    <!-- All heat runs and WiX Fragment files containing Component elements will reference this ComponentGroup -->
-    <ComponentGroup Id="ProductComponentGroup" Directory="INSTALLDIR"/>
```
Is there a reason this is getting removed? I don't see any explanation in the commits

jcoconnor replied:
Shouldn't be removed - will fix.

melissa commented:
:+1: pending my question about the line removal in `resources/windows/wix/project.wxs.erb`

ferventcoder commented:
this is a test @here
